lib/mlir/Dialect/Clift/IR/CliftOps.cpp

```diff
@@ -25,6 +25,42 @@ void mlir::clift::CliftDialect::registerOperations() {
                 /* End of operations list */>();
 }
 
+//===---------------------------- Region types ----------------------------===//
+
+template<typename OpInterface>
+static bool verifyRegionContent(Region &R, const bool Required) {
```

Comment:
It's a regular parameter of `verifyExpressionRegion`. I see no reason to jump through hoops to make it a template parameter of `verifyRegionContent`.

Reply:
This is really not a big deal. But I'm trying to understand what are the limits of what we can do with MLIR, TableGen and ODS.

You say that `Required` is a regular parameter of `verifyExpressionRegion` and that's true.
But my understanding is that it's only so because of the following `TableGen` lines in `CliftOps.td`:
```
def Clift_ExpressionRegion : Region<CPred<"::mlir::clift::impl::verifyExpressionRegion($_self, true)">,
                                    "Region representing an expression">;

def Clift_OptionalExpressionRegion : Region<CPred<"::mlir::clift::impl::verifyExpressionRegion($_self, false)">,
                                            "Optional region representing an expression">;
```

Could we change those line to the following? (now with a template argument for `verifyExpressionRegion`)
```
def Clift_ExpressionRegion : Region<CPred<"::mlir::clift::impl::verifyExpressionRegion<true>($_self)">,
                                    "Region representing an expression">;

def Clift_OptionalExpressionRegion : Region<CPred<"::mlir::clift::impl::verifyExpressionRegion<false>($_self)">,
                                            "Optional region representing an expression">;
```

If so, I would prefer it.

Reply:
We could, but it requires either moving the functions into the header entirely, or explicitly declaring and instantiating the template for both true and false. I can do that if you prefer, but like I said, it's more hoops to jump through. I personally don't see any benefit in making it a template parameter.

Reply:
No big deal, let's leave it like this.

Thanks for the explanation.